src/wfs/src/main/java/org/geoserver/wfs/WFSTransactionExceptionFactory.java

```diff
@@ -0,0 +1,74 @@
+/* (c) 2023 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wfs;
+
+import org.geoserver.config.SettingsInfo;
+
+/**
+ * Factory for creating WFS_T application specific exception instances, including cause details in
+ * exception messages based on the global verbose exceptions setting. This enables eg trigger
+ * exception messages to bubble up to end-users in a controlled internal environment, where root
+ * cause aids the user in entering correct data and is of no security concern.
+ *
+ * @author Martin Kalén
+ */
+public class WFSTransactionExceptionFactory {
```

Comment:
Seems to be quite specific to the task at hand, I'd make it package private while you're at it. As a general direction, we're trying not to make new classes public, unless they need to be.

Reply:
Check. Now package private.